chore: Property filter custom forms input dropdown

pan-kot opened the pull request:
### Description

Updates property filter so that tokens with custom forms have their value editable with a custom input dropdown to ensure the value input takes less space.

Rel: Nu26A4Kr5E80

### How has this been tested?

* Unit tests and manual testing
* Consultation with accessibility team

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

connorlanigan commented:
The dropdown trigger for the custom property value behaves inconsistently with the dropdown triggers for the property name and the operator. When I click the trigger to open the dropdown and then click the trigger again, the dropdown stays open. However, it should close.

Another inconsistency is with the keyboard interaction: when I focus the custom property value trigger and press Space, the dropdown opens. When I then press Shift-Tab to move to the operator trigger, the dropdown of the custom property value stays open. For the other Select elements, the dropdown closes in that case. I recommend that we align this with the behaviour of the Date Range Picker component, which traps the focus in the dropdown.

I wonder if we can potentially also reuse logic here from other components, e.g. [the Select component's `useSelect` hook](https://github.com/cloudscape-design/components/blob/main/src/select/utils/use-select.ts#L166-L185), which also addresses focus handling and keyboard interaction.

pan-kot commented:
> The dropdown trigger for the custom property value behaves inconsistently with the dropdown triggers for the property name and the operator. When I click the trigger to open the dropdown and then click the trigger again, the dropdown stays open. However, it should close.

When there is a custom form we can consider it a different type of an input. By default the value is edited with an autosuggest but if, say, we added a date picker instead there will be the same inconsistency.

> Another inconsistency is with the keyboard interaction: when I focus the custom property value trigger and press Space, the dropdown opens. When I then press Shift-Tab to move to the operator trigger, the dropdown of the custom property value stays open. For the other Select elements, the dropdown closes in that case. I recommend that we align this with the behaviour of the Date Range Picker component, which traps the focus in the dropdown.

Thanks for that observation. Turns out the tab trap that we already have inside the dropdown is not enough. I added a focus lock same as in date- and date-range pickers.

> I wonder if we can potentially also reuse logic here from other components, e.g. [the Select component's `useSelect` hook](https://github.com/cloudscape-design/components/blob/main/src/select/utils/use-select.ts#L166-L185), which also addresses focus handling and keyboard interaction.

I think the select/autosuggest components are very different. E.g. for the custom form dropdown there are explicit Cancel/Apply buttons and more.

pan-kot commented:
Closed in favour of Qeb3AbRK3bNw
